static/js/subs.js

```diff
@@ -563,11 +570,37 @@ exports.setup_page = function (callback) {
         exports.actually_filter_streams();
         stream_create.set_up_handlers();
 
+        $("#add_new_subscription_extend input[type='text']").on("input", function () {
+            // Debounce filtering in case a user is typing quickly
```

Comment:
It looks like this comment was copy/pasted from another function, but I don't understand what it means in either place.  Is this supposed to be a TODO?  Can you try to do `git blame` on the other place we have this comment to figure out the intention here?

Reply:
The purpose of the event handler is of course to initiate the filtering of streams whenever an input is made to those text boxes. The event handler was already present, as on lines `578` to `581` and I only had to add a similar handler for the other search box I added (lines `573` to `576`) because we want filtering to take place if the user types in either box, hence we want to capture both the events.

Coming to the comment:

A `git blame` on the original handler reveals:
```(shell)
1886f0a015c (Brock Whittaker 2016-11-01 14:32:10 -0700 578)         $("#add_new_subscription input[type='text']").on("input", function () {
fe2d38392f0 (Tim Abbott      2016-12-14 22:36:30 -0800 579)             // Debounce filtering in case a user is typing quickly
fe2d38392f0 (Tim Abbott      2016-12-14 22:36:30 -0800 580)             filter_streams();
1886f0a015c (Brock Whittaker 2016-11-01 14:32:10 -0700 581)         });
```

If you checkout fe2d38392f01e17b8a0a8da6871bf5f41d3474ea, you'll see that it changes a direct call to `actually_filter_streams` into a call via `_.throttle` (which is what `filter_streams` does) in order to ignore calls happening very quickly. Hence the "debouncing".
So, it is not a TODO, but something that is being done in that function.

Because it is unrelated, I'll save changes that clarify this comment for a separate commit.